### PR TITLE
[global] update requirements checking flow

### PR DESCRIPTION
**Pull request by yalosev**
Signed-off-by: Yuriy Losev <yuriy.losev@flant.com>

## Description
Rework release requirements check

## Why do we need it, and what problem does it solve?
Values reading is not allowed from the other module. So, we have to separate requirements check mechanics and check them explicitly.

## What is the expected result?
```
v1-66-0   false      Pending    0s               "ingressNginx" requirement for DeckhouseRelease "1.66.0" not met: minimal IngressNginxController version is lower then required
```

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: global
type: fix 
summary: Reworked flow that checks release requirements.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/402-ingress-nginx/requirements/check_test.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 Flant JSC
+Copyright 2022 Flant JSC
```
Comment (nabokihms) [inline modules/402-ingress-nginx/requirements/check_test.go:2]: When you are moving files, changing the copyright is unnecessary. The copyright date means the creation date. If we can see in the git that a file was moved, it is good enough.

**Review by nabokihms — APPROVED** [review]

File: modules/402-ingress-nginx/requirements/check_test.go (hunk 1 above)
Comment (nabokihms) [reply modules/402-ingress-nginx/requirements/check_test.go:2]: You can resolve the comment when you read it. Changing this is also not required.

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [2820](https://github.com/deckhouse/deckhouse/pull/2820) to the branch [release-1.38](https://github.com/deckhouse/deckhouse/tree/release-1.38) successful!

**Pull request merged**